storage/netapp/netapp_e_volume.py

```diff
@@ -0,0 +1,621 @@
+                                     url_username=self.api_usr, url_password=self.api_pwd,
+                                     validate_certs=self.validate_certs, timeout=120)
+            except Exception:
+                err = get_exception()
+                self.module.fail_json(
+                    msg="Failed to expand thin volume.  Volume [%s].  Array Id [%s]. Error[%s]." % (self.name,
+                                                                                                    self.ssid,
+                                                                                                    str(err)))
+
+                # TODO: check return code
+        else:
+            self.debug('expanding volume')
+            volume_expand_req = dict(
+                expansionSize=self.size,
+                sizeUnit=self.size_unit
+            )
+            try:
+                (rc, resp) = request(
+                    self.api_url + "/storage-systems/%s/volumes/%s/expand" % (self.ssid,
+                                                                              self.volume_detail['id']),
+                    data=json.dumps(volume_expand_req), headers=self._post_headers, method='POST',
+                    url_username=self.api_usr, url_password=self.api_pwd, validate_certs=self.validate_certs,
+                    timeout=120)
+            except Exception:
+                err = get_exception()
+                self.module.fail_json(
+                    msg="Failed to expand volume.  Volume [%s].  Array Id [%s]. Error[%s]." % (self.name,
+                                                                                               self.ssid,
+                                                                                               str(err)))
+
+            self.debug('polling for completion...')
+
+            while True:
+                try:
+                    (rc, resp) = request(self.api_url + "/storage-systems/%s/volumes/%s/expand" % (self.ssid,
+                                                                                                   self.volume_detail[
+                                                                                                       'id']),
+                                         method='GET', url_username=self.api_usr, url_password=self.api_pwd,
+                                         validate_certs=self.validate_certs)
+                except Exception:
+                    err = get_exception()
+                    self.module.fail_json(
+                        msg="Failed to get volume expansion progress.  Volume [%s].  Array Id [%s]. Error[%s]." % (
+                            self.name, self.ssid, str(err)))
+
+                action = resp['action']
+                percent_complete = resp['percentComplete']
+
+                self.debug('expand action %s, %s% complete...' % (action, percent_complete))
```

Comment:
The %% was another error in string format conversion tool.  It's fixed.

For small modules like these, I don't like adding a dedicated logger.  We should be able to give the user a real good idea about what is wrong with simple fail_json.   In these mods the debug logging is essentially technical debt since they appear to be developer debugging statements.  Tweaking the argument_spec and adding a command line arg allows me to debug a module.  This is preferable to print statements.  I want to remove the logging from all the modules and add debuggin instructions to README.md.